app/src/main/java/org/wikipedia/feed/onthisday/OnThisDayFragment.java

```diff
@@ -88,83 +97,81 @@ public static OnThisDayFragment newInstance(int age, WikiSite wikiSite) {
     }
 
     @Override
+    @SuppressWarnings("checkstyle:magicnumber")
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_on_this_day, container, false);
         unbinder = ButterKnife.bind(this, view);
         int age = requireActivity().getIntent().getIntExtra(AGE, 0);
         wiki = requireActivity().getIntent().getParcelableExtra(WIKISITE);
         date = DateUtil.getDefaultDateFor(age);
+        yearClickedOnCardView = requireActivity().getIntent().getIntExtra(YEAR, -1);
         setUpToolbar();
         eventsRecycler.setLayoutManager(new LinearLayoutManager(getContext(), RecyclerView.VERTICAL, false));
 
         final int topDecorationDp = 24;
         eventsRecycler.addItemDecoration(new HeaderMarginItemDecoration(topDecorationDp, 0));
         setUpRecycler(eventsRecycler);
-
         errorView.setBackClickListener(v -> requireActivity().finish());
 
-        if (requireActivity().getWindow().getSharedElementEnterTransition() != null
-                && savedInstanceState == null) {
-            final int animDelay = 500;
-            dayText.postDelayed(() -> {
-                if (!isAdded() || dayText == null) {
-                    return;
-                }
-                updateContents(age);
-                ValueAnimator colorAnimation = ValueAnimator.ofObject(new ArgbEvaluator(), dayText.getCurrentTextColor(),
-                        ResourceUtil.getThemedColor(requireContext(), R.attr.material_theme_primary_color));
-                colorAnimation.addUpdateListener(animator -> {
-                    if (dayText != null) {
-                        dayText.setTextColor((Integer) animator.getAnimatedValue());
-                    }
-                });
-                colorAnimation.start();
-            }, animDelay);
-        } else {
-            dayText.setTextColor(ResourceUtil.getThemedColor(requireContext(), R.attr.material_theme_primary_color));
+        final int animDelay = (requireActivity().getWindow().getSharedElementEnterTransition() != null
+                && savedInstanceState == null) ? 500 : 0;
+        onThisDayTitleView.postDelayed(() -> {
+            if (!isAdded() || onThisDayTitleView == null) {
+                return;
+            }
```

Comment:
A little bit messy to me if we put the code here. Would it be better if create a method of it? (and we can move the suppress `checkstyle` to the method.